sharding/contracts/sharding_manager_test.go

```diff
@@ -1,130 +1,560 @@
+	notaryLockupLenght   *big.Int
+	proposerLockupLength *big.Int
+}
+
 var (
-	key, _                   = crypto.HexToECDSA("b71c71a67e1177ad4e901695e1b4b9ee17ae16c6668d313eac2f96dbcda3f291")
-	addr                     = crypto.PubkeyToAddress(key.PublicKey)
-	accountBalance1001Eth, _ = new(big.Int).SetString("1001000000000000000000", 10)
-	collatorDeposit, _       = new(big.Int).SetString("1000000000000000000000", 10)
+	mainKey, _                   = crypto.HexToECDSA("b71c71a67e1177ad4e901695e1b4b9ee17ae16c6668d313eac2f96dbcda3f291")
+	accountBalance2000Eth, _     = new(big.Int).SetString("2000000000000000000000", 10)
+	notaryDepositInsufficient, _ = new(big.Int).SetString("999000000000000000000", 10)
+	notaryDeposit, _             = new(big.Int).SetString("1000000000000000000000", 10)
+	smcConfig                    = SMCConfig{
+		notaryLockupLenght:   new(big.Int).SetInt64(1),
+		proposerLockupLength: new(big.Int).SetInt64(1),
+	}
+	ctx = context.Background()
 )
 
-func deploySMCContract(backend *backends.SimulatedBackend) (common.Address, *types.Transaction, *SMC, error) {
+func deploySMCContract(backend *backends.SimulatedBackend, key *ecdsa.PrivateKey) (common.Address, *types.Transaction, *SMC, error) {
 	transactOpts := bind.NewKeyedTransactor(key)
 	defer backend.Commit()
 	return DeploySMC(transactOpts, backend)
 }
 
 // Test creating the SMC contract
 func TestContractCreation(t *testing.T) {
-	backend := backends.NewSimulatedBackend(core.GenesisAlloc{addr: {Balance: accountBalance1001Eth}})
-	_, _, _, err := deploySMCContract(backend)
+	addr := crypto.PubkeyToAddress(mainKey.PublicKey)
+	backend := backends.NewSimulatedBackend(core.GenesisAlloc{addr: {Balance: accountBalance2000Eth}})
+	_, _, _, err := deploySMCContract(backend, mainKey)
 	backend.Commit()
 	if err != nil {
 		t.Fatalf("can't deploy SMC: %v", err)
 	}
 }
 
-// Test getting the collation gas limit
-func TestGetCollationGasLimit(t *testing.T) {
-	backend := backends.NewSimulatedBackend(core.GenesisAlloc{addr: {Balance: accountBalance1001Eth}})
-	_, _, smc, _ := deploySMCContract(backend)
-	gasLimit, err := smc.GetCollationGasLimit(&bind.CallOpts{})
+func TestNotaryRegister(t *testing.T) {
+	const notaryCount = 3
+	var notaryPoolAddr [notaryCount]common.Address
+	var notaryPoolPrivKeys [notaryCount]*ecdsa.PrivateKey
+	var txOpts [notaryCount]*bind.TransactOpts
+	genesis := make(core.GenesisAlloc)
+
+	// initializes back end with 3 accounts and each with 2000 eth balances
+	for i := 0; i < notaryCount; i++ {
+		key, _ := crypto.GenerateKey()
+		notaryPoolPrivKeys[i] = key
+		notaryPoolAddr[i] = crypto.PubkeyToAddress(key.PublicKey)
+		txOpts[i] = bind.NewKeyedTransactor(key)
+		txOpts[i].Value = notaryDeposit
+		genesis[notaryPoolAddr[i]] = core.GenesisAccount{
+			Balance: accountBalance2000Eth,
+		}
+	}
+
+	backend := backends.NewSimulatedBackend(genesis)
+	_, _, smc, _ := deploySMCContract(backend, notaryPoolPrivKeys[0])
+
+	// Test notary 0 has not registered
```

Comment:
What does this mean? You are testing that notary 0 has not been registered?

Reply:
Yes. it's a negative test case where notary 0's not registered